src/modules/CRUD/components/Details/Read/GeomView/GeomView.js

```diff
@@ -0,0 +1,255 @@
+      coordinates: getUISchemaCoordinates(coordinates),
+      type: {
+        'ui:widget': 'hidden',
+      },
+    },
+    title,
+    type: 'object',
+    value,
+  };
+};
+
+const getProperties = geometries => Object.entries(geometries).reduce(
+  (acc, [key, value], index) => ({ ...acc, [key]: getJSONSchemaFromGeom(value, index) }),
+  {},
+);
+
+const updateGeometry = ({
+  features: [{ geometry, id }],
+  geomType,
+  geometries,
+  name,
+  setGeomValues,
+  target: map,
+  type,
+}) => {
+  if (type === 'draw.delete') {
+    setGeomValues(getProperties({
+      ...geometries,
+      [name]: {
+        ...geometries[name],
+        geom: null,
+      },
+    }));
+    return;
+  }
+
+  const { features } = map.draw.getAll();
+
+  if (features.length > 1 && [POINT, LINESTRING, POLYGON].includes(geomType)) {
+    map.draw.delete(
+      features.reduce((list, feature) => (
+        feature.id !== id
+          ? [...list, feature.id]
+          : list
+      ), []),
+    );
+  }
+
+  setGeomValues(getProperties({
+    ...geometries,
+    [name]: {
+      ...geometries[name],
+      geom: geometry,
+    },
+  }));
+};
+
+const initDraw = ({
+  addControl,
+  layerPaintId,
+  map,
+  name,
+  geometries,
+  setGeomValues,
+}) => {
+  const { [name]: { geom, identifier, geom_type: geomType } } = geometries;
+  if (!Object.keys(map).length) {
+    return;
+  }
+
+  const handleUpdateGeometry = event => (
+    updateGeometry({ ...event, geomType, geometries, name, setGeomValues })
+  );
+
+  const control = {
+    control: CONTROL_DRAW,
+    position: CONTROLS_TOP_LEFT,
+    onDrawUpdate: handleUpdateGeometry,
+    onDrawCreate: handleUpdateGeometry,
+    onDrawDelete: handleUpdateGeometry,
+    controls: {
+      point: [ALL, POINT, MULTI_POINT].includes(geomType),
+      line_string: [ALL, LINESTRING, MULTI_LINESTRING].includes(geomType),
+      polygon:  [ALL, POLYGON, MULTI_POLYGON].includes(geomType),
+      trash: true,
+      combine_features: false,
+      uncombine_features: false,
+    },
+    order: 2,
+  };
+
+  if (geom !== null) {
+    const onControlAdded = ({ control: addedControl }) => {
+      if (addedControl !== control.control) {
+        return;
+      }
+      map.setFilter(layerPaintId, ['!=', '_id', identifier]);
+      map.draw.add(geom);
+      map.off('control_added', onControlAdded);
+    };
+    map.on('control_added', onControlAdded);
+  }
+  addControl(control);
+};
+
+const getLayerPaintId = (layerPaints, name) => {
+  const { id } = layerPaints.find(({ 'source-layer': source }) => source === name) || {};
+  return id;
+};
+
+const resetStyle = (map, layerPaints, name) => {
+  if (!map) {
+    return;
+  }
+  const layerPaintId = getLayerPaintId(layerPaints, name);
+  map.setFilter(layerPaintId, null);
+};
+
+const GeomView = ({
+  addControl,
+  layerPaints,
+  map,
+  removeControl,
+  geometries,
+}) => {
+  const [geomValues, setGeomValues] = useState(getProperties(geometries));
```

Comment:
UseMemo here also ?

Reply:
I need `setGeomValues` method elsewhere of the mount. But maybe you can help me to improve this part.

Reply:
Ok i didn't see it.